BLSQ/iaso
# WC2-569 Add a feature flag on the entity duplicate merging that allows (soft-)deletion of A or B

This PR allows the merge entity duplicate functionality to change to a "(soft) delete one of the 2".

Changes:
- Add account feature flag ENTITY_DUPLICATES_SOFT_DELETE to change
behavior on the entity duplicate details page from a merge action to
allow a soft deletion of either.
- Implement the (soft) delete action on DELETE /api/entities/pk. This
action will:
  - Soft delete the entity
  - Soft delete it's attributes + instances
  - Delete any attached pending `EntityDuplicate` pairs

Related JIRA tickets : https://bluesquare.atlassian.net/browse/WC2-569


## How to test

Have an account with some entity duplicates. (e.g. with the setuper)
When you set/unset the new feature flag `ENTITY_DUPLICATES_SOFT_DELETE` on your account and you should see the page change.

## Print screen / video

### Regular merge screen
![2024-10-01_22-34](https://github.com/user-attachments/assets/8d62eecc-d652-4de4-810c-d20052d8b8b3)


### With feature flag activated
![2024-10-01_22-51](https://github.com/user-attachments/assets/dc2122bc-12a0-41f9-b52c-9c3740ca6861)

### The feature flag in the Django admin
![2024-10-01_22-35](https://github.com/user-attachments/assets/ecd24c8d-9a28-46fd-8986-138d428b7ee7)
